docs/tutorials/training_transformers.md

```diff
@@ -0,0 +1,93 @@
+
+```python
+import json
+import robomimic
+import robomimic.utils.torch_utils as TorchUtils
+from robomimic.config import config_factory
+from robomimic.scripts.train import train
+
+# load json with default transformer parameters
+ext_cfg = json.load(open('robomimic/config/default_templates/bc_transformer.json', 'r'))
+#generate base behavioral cloning config
+config = config_factory(ext_cfg["algo_name"])
+
+with config.values_unlocked():
+    # update config with default transformer parameters
+    config.update(ext_cfg)
+
+    # set config attributes for training
+    config.experiment.name = "bc_transformer_example"
+    config.train.data = "/path/to/dataset.hdf5"
+    config.train.output_dir = "/path/to/output_dir/"
+
+    # set horizon to be greater than the length of your longest demonstration
+    config.experiment.rollout.horizon = 700
+
+    # set config attributes for image modalities
+    config.experiment.save.every_n_epochs = 20
+    config.experiment.rollout.rate = 20
+    config.experiment.epoch_every_n_steps = 500
+    config.experiment.validation_epoch_every_n_steps = 50
+
+    config.train.batch_size = 16
+    config.train.num_epochs = 600
+    config.train.num_data_workers = 2
+
+    # Keep low-dimensional states for robot arm position/orientation
+    config.observation.modalities.obs.low_dim = [
+                        "robot0_eef_pos",
+                        "robot0_eef_quat",
+                        "robot0_gripper_qpos"
+                    ]
+    # Add images to observation, names may depend on your dataset naming convention
+    config.observation.modalities.obs.rgb = [
+                        "agentview_image",
+                        "robot0_eye_in_hand_image"
+                    ]
+
+    #set config attributes for transformer
+    config.algo.transformer.embed_dim = 512                       # dimension for embeddings used by transformer
+    config.algo.transformer.num_layers = 6                        # number of transformer blocks to stack
+    config.algo.transformer.num_heads = 8                         # number of attention heads for each transformer block (should divide embed_dim evenly)
+    config.algo.transformer.context_length = 10                   # length of (s, a) seqeunces to feed to transformer
+    config.train.frame_stack = 10                                 # shoule be same as context length
+    config.train.seq_length = 1                                   # length of (s, a) seqeunces for transformer to predict
+
+# get torch device
+device = TorchUtils.get_torch_device(try_to_use_cuda=True)
+
+# launch training run
+train(config, device=device)
+```
+
+Optionally, you can create a copy of `robomimic/config/default_templates/bc_transformer.json` and store it locally on your computer.  From here you can directly modify the file and call `train.py` as in **Section 1** with the `--config` argument set to `/path/to/local_bc_transformer.json`.
```

Comment:
Can we suggest using the [hyperparameter helper](https://robomimic.github.io/docs/tutorials/hyperparam_scan.html#step-1-create-base-config-json) as the preferred way to set custom settings, and suggest copying `robomimic/config/default_templates/bc_transformer.json` to a new directory to use as the base json? Or at the very least suggest this alternative. What do you think @snasiriany ? This is how I would use the default transformer config to generate new configs for custom training use cases.

Reply:
I think this section on how to modify / update configs isn't related to transformers per se. I'd just defer readers to the [configs](https://robomimic.github.io/docs/tutorials/configs.html) and [hyperparams](https://robomimic.github.io/docs/tutorials/hyperparam_scan.html) tutorials. That way we can keep this tutorial section light-weight.

Re copying to a new directory as the base json. Yes, I do this. I copy to a new json and then modify on top of that with the hyperparam generator. But again, this isn't related to transformers per se. So I'd modify the config/hyperparams tutorial pages to explain this

Reply:
I agree with this @MBronars

Reply:
@amandlek you mean you agree with me?